addons/binding/org.openhab.binding.upb/src/main/java/org/openhab/binding/upb/internal/UPBMessage.java

```diff
@@ -0,0 +1,309 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.upb.internal;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map.Entry;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Model for a message sent or received from a UPB modem.
+ *
+ * @author Chris Van Orman
+ * @since 1.9.0
+ */
+public class UPBMessage {
+
+    /**
+     * An enum of possible message priorities.
+     * 
+     * @author Chris
+     *
+     */
+    public enum Priority {
+        LOW(0, 1000),
+        NORMAL(1, 0);
+
+        private int value;
+        private int delay;
+
+        private Priority(int value, int delay) {
+            this.value = value;
+            this.delay = delay;
+        }
+
+        /**
+         * @return the value
+         */
+        public int getValue() {
+            return value;
+        }
+
+        /**
+         * @return the delay
+         */
+        public int getDelay() {
+            return delay;
+        }
+    }
+
+    /**
+     * An enum of possible commands.
+     *
+     * @author Chris Van Orman
+     *
+     */
+    public enum Command {
+        ACTIVATE,
```

Comment:
You should store the values in your enums, that is prefered over this way, see for example [the rfxcom binding](https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComLighting1Message.java#L35) 

Integrating the fetching of actual enum values based on an value is also shown there